pkgs/development/libraries/gtk/4.x.nix

```diff
@@ -165,6 +162,11 @@ stdenv.mkDerivation (finalAttrs: {
     # Required for GSettings schemas at runtime.
     # Will be picked up by wrapGAppsHook4.
     gsettings-desktop-schemas
+  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    (darwinMinVersionHook "15.0")
+  ] ++ lib.optionals (stdenv.hostPlatform.isDarwin
+                   && lib.versionOlder apple-sdk.version "10.15") [
+      apple-sdk_10_15
```

Comment:
not sure if this logic is totally correct as it seems like a little hack to assume that the current sdk is `apple-sdk` cause it could be propagated by some other package or something.

but this still should be good enough as is limited to just x64 Darwin.

Reply:
It should be fine, since newer SDKs win: the only reason to conditionalize this is because `apple-sdk_10_15` can fail to evaluate (on `aarch64-darwin`), which is safely covered by the case `lib.versionOlder apple-sdk.version "10.15"`.